internal_coroutines_test.go

```diff
@@ -1034,3 +1034,53 @@ func TestDecodeFutureChain(t *testing.T) {
 	}
 	require.EqualValues(t, expected, history)
 }
+
+func TestSelectFuture_WithBatchSets(t *testing.T) {
+	var history []string
+	d := newDispatcher(background, func(ctx Context) {
+		future1, settable1 := NewFuture(ctx)
+		future2, settable2 := NewFuture(ctx)
+		future3, settable3 := NewFuture(ctx)
+
+		s := NewSelector(ctx)
```

Comment:
Select is about reading. Future can be read multiple times as opposed to a channel. So I don't see a problem calling each callback once for a future that is ready.